File: rest-api-jsonplaceholder/APIUser.java
///usr/bin/env jbang "$0" "$@" ; exit $?

//DEPS io.github.wfouche.tulip:tulip-runtime:2.2.3
//JAVA 25
//RUNTIME_OPTIONS -Xmx512m -XX:+UseZGC

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import io.github.wfouche.tulip.api.TulipApi;
import io.github.wfouche.tulip.user.HttpUser;

public class APIUser extends HttpUser {

    // Action 1: GET /posts
    public boolean action1() {
        var response = httpGet("/posts");
        return response.isSuccessful();
    }

    // Action 2: POST /posts
    public boolean action2() {
        String payload = """
        {
            "title": "Tulip Test Post",
            "body": "This is a test post created during load testing",
            "userId": 1
        }       
        """;
        var response = httpPost(payload, "/posts");
        return response.isSuccessful();
    }

    // Action 3: GET /posts/{id}
    public boolean action3() {
        int post_id = 1;
        var response = httpGet("/posts/{id}", post_id);
        return response.isSuccessful();
    }

    public Logger logger() {
        return logger;
    }

    // Logger
    private static final Logger logger = LoggerFactory.getLogger(APIUser.class);

    // Main method to run the Tulip load test
    public static void main(String[] args) {
        TulipApi.runTulip("benchmark_config.json");
    }

}

File: rest-api-jsonplaceholder/APIUser.kt
//KOTLIN 2.2.21

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import io.github.wfouche.tulip.user.HttpUser

class APIUser : HttpUser() {
    // Action 1: GET /posts
    override fun action1(): Boolean {
        val response = httpGet("/posts")
        return response.isSuccessful()
    }

    // Action 2: POST /posts
    override fun action2(): Boolean {
        val payload: String = """
        {
            "title": "Tulip Test Post",
            "body": "This is a test post created during load testing",
            "userId": 1
        }       
        
        """.trimIndent()
        val response = httpPost(payload, "/posts")
        return response.isSuccessful()
    }

    // Action 3: GET /posts/{id}
    override  fun action3(): Boolean {
        val post_id = 1
        val response = httpGet("/posts/{id}", post_id)
        return response.isSuccessful()
    }

    override  fun logger(): Logger {
        return elog
    }

    companion object {
        // Logger
        private val elog: Logger = LoggerFactory.getLogger(APIUser::class.java)
    }
}